projects/snapshot_bot/config.cc

```diff
@@ -0,0 +1,271 @@
+#include "config.h"
+
+// BoB robotics includes
+#include "common/macros.h"
+#include "common/path.h"
+
+// Standard C++ includes
+#include <limits>
+
+using namespace BoBRobotics;
+using Milliseconds = std::chrono::duration<double, std::milli>;
+
+namespace {
+// **YUCK** as well as std::strings, it also seems like OpenCV can't serialise size_t
+int getIntegerSize(size_t value)
```

Comment:
you could just accept that the OpenCV serialization sucks somewhat, use ``int``/``unsigned int`` and ``static_cast`` when required..